modules/cuda_plugin/tests/unit/cuda/stl/algorithms/sort.cu

```diff
@@ -0,0 +1,226 @@
+    auto end = first + (blockIdx.x + 1) * chunk_size;
+    if (end > last) {
+        end = last;
+    }
+    auto stack_begin = stack + blockIdx.x * chunk_size;
+    constexpr auto kNumPartitions = 1024;
+    CUDA::algorithms::parallel::quick_sort_iterative<kNumPartitions>(
+        begin, end, stack_begin, comparer, threadIdx.x, blockDim.x);
+}
+
+template <typename T, typename Compare>
+__global__ void parallel_quick_sort_launch(T* first, T* last, Compare comparer) {
+    const auto size = last - first;
+    const auto num_chunks = (size % gridDim.x == 0) ? gridDim.x : (gridDim.x + 1);
+    const auto chunk_size = size / num_chunks;
+    auto begin = first + blockIdx.x * chunk_size;
+    auto end = first + (blockIdx.x + 1) * chunk_size;
+    if (end > last) {
+        end = last;
+    }
+    constexpr auto kNumPartitions = 1024;
+    CUDA::algorithms::parallel::quick_sort_iterative<kNumPartitions>(begin, end, comparer, threadIdx.x, blockDim.x);
+}
+
+}  // namespace
+
+class QuickSortKernelTest : public testing::Test {
+    void SetUp() override {}
+
+    void TearDown() override {}
+
+public:
+    template <size_t ArrayLength, bool WithStack>
+    void runQuickSort() {
+        auto arr = std::make_unique<float[]>(ArrayLength);
+        auto sortedArr = std::make_unique<float[]>(ArrayLength);
+        std::random_device rd{};
+        std::mt19937 gen{rd()};
+        std::uniform_real_distribution<float> dist(-20000000, 20000000);
+
+        constexpr size_t arr_size = sizeof(float) * ArrayLength;
+
+        for (int n = 0; n < ArrayLength; ++n) {
+            arr[n] = dist(gen);
+        }
+
+        const auto& defaultStream = CUDA::DefaultStream::stream();
+        auto src = defaultStream.malloc(arr_size);
+        defaultStream.upload(src, arr.get(), arr_size);
+        auto begin = static_cast<float*>(src.get());
+        auto end = begin + ArrayLength;
+        CUDA::Stream stream{};
+        if (WithStack) {
+            auto stack = defaultStream.malloc(arr_size);
+            auto stack_begin = static_cast<int*>(stack.get());
+            quick_sort_launch<<<1, 1, 0, stream.get()>>>(begin, end, stack_begin, CUDA::algorithms::Less<float>{});
+        } else {
+            quick_sort_launch<<<1, 1, 0, stream.get()>>>(begin, end, CUDA::algorithms::Less<float>{});
+        }
+        stream.synchronize();
+        defaultStream.download(sortedArr.get(), src, arr_size);
+
+        std::sort(arr.get(), arr.get() + ArrayLength);
+        for (int i = 0; i < ArrayLength; ++i) {
+            if (sortedArr[i] != arr[i]) {
+                std::cout << "i = " << i << std::endl;
+                std::cout << "arr[i] = " << arr[i] << std::endl;
+                std::cout << "sortedArr[i] = " << sortedArr[i] << std::endl;
+            }
+            ASSERT_FLOAT_EQ(sortedArr[i], arr[i]);
+        }
+    }
+
+    template <size_t ArrayLength, size_t TopK, bool WithStack>
+    void runPartialQuickSort() {
```

Comment:
runPartialQuickSort & runParallelQuickSort
differ only by:
calling either parallel_quick_sort_launch or quick_sort_launch. The rest code could be shared here.